client/blocks/comment-button/index.jsx

```diff
@@ -39,7 +45,7 @@ const CommentButton = React.createClass( {
 	render() {
 		let label;
 		const containerTag = this.props.tagName,
-			commentCount = this.props.commentCount;
+			commentCount = ( this.props.syncedCount === undefined ? this.props.commentCount : this.props.syncedCount );
```

Comment:
> Shouldn’t we include the syncedCount prop in the propTypes?

Yes, we should :), that was an omission error on my part, and I'll add it in my next commit.

> Actually, why do we need both commentCount and syncedCount?

For now we need both for slightly complicated reasons:

- `commentCount` is the prop passed in when using CommentButton, and it's the default comment count we'll display
- `syncedCount` is the redux state's current comment count for the post. It can be undefined if the list of comments has not yet been fetched for the post, which is why we need a default.

The total comment count could be fetched separately when rendering CommentButton, but here's why I decided against it:

- we could fetch it using the API endpoint `/posts/$post_ID/replies`, but it's an extra network request
- we can't get it from comment redux state, because it's unavailable until we [fetch all comments](https://github.com/Automattic/wp-calypso/blob/master/client/state/comments/actions.js#L61)
- fetching all comments would also mean another network request
- we do know the comment count when displaying a post, but it's stored in flux, and can't neatly be added to redux's state
- therefore, to reduce # of network requests, we pass in the comment count we know from the post

In the future, when we're using redux to store post state, we can populate the redux `state.comments.totalCommentCount`, and remove the `commentCount` prop